app/routes/companyInterest/components/CompanyInterestList.tsx

```diff
@@ -84,11 +115,61 @@ class CompanyInterestList extends Component<Props, State> {
         },
       })
       .then(() => {
-        this.props.push(`/companyInterest?semesters=${clickedOption.id}`);
+        this.props.push(
+          `/companyInterest?semesters=${clickedOption.id}&event=${this.state.selectedEvent}`
+        );
       });
   };
+  handleEventChange = (clickedOption: EventOption): void => {
+    const { value } = clickedOption;
+    this.props
+      .fetch({
+        filters: {
+          events: value !== null ? value : null,
+        },
+      })
+      .then(() => {
+        this.props.push(
```

Comment:
Please use `replace`, otherwise this will fill up the browser history

Reply:
Done.